pyccel/ast/numpyext.py

```diff
@@ -532,12 +593,8 @@ def __init__(self, shape, fill_value, dtype=None, order='C'):
         order = NumpyNewArray._process_order(order)
 
         # Cast fill_value to correct type
-        # TODO [YG, 09.11.2020]: treat difficult case of LiteralComplex
-        from pyccel.ast.datatypes import str_dtype
-        stype = str_dtype(dtype)
-        if fill_value and stype != 'complex':
-            from pyccel.codegen.printing.fcode import python_builtin_datatypes
-            cast_func  = python_builtin_datatypes[stype]
+        if fill_value and not isinstance(fill_value, Nil):
```

Comment:
We seem to have a lot of places in this file where we check `if X is not None and not isinstance(X, Nil)`, it may be beyond the scope of this PR but I think it would be nice to add a magic method `__bool__` to `Nil` which returns `False` then all these tests can be replaced with `if X`. What do you think?

Reply:
I can do that in another PR.

Reply:
ok :+1:

Reply:
Good idea!